components/profile_popover/profile_popover.tsx

```diff
@@ -16,263 +13,270 @@ import {browserHistory} from 'utils/browser_history';
-
-        currentTeamId: PropTypes.string.isRequired,
-
-        /**
-         * @internal
-         */
-        currentUserId: PropTypes.string.isRequired,
-        customStatus: PropTypes.object,
-        isCustomStatusEnabled: PropTypes.bool.isRequired,
-
-        /**
-         * @internal
-         */
-        hasMention: PropTypes.bool,
-
-        /**
-         * @internal
-         */
-        isInCurrentTeam: PropTypes.bool.isRequired,
-
-        /**
-         * @internal
-         */
-        teamUrl: PropTypes.string.isRequired,
-
-        /**
-         * @internal
-         */
-        isTeamAdmin: PropTypes.bool.isRequired,
-
-        /**
-         * @internal
-         */
-        isChannelAdmin: PropTypes.bool.isRequired,
-
-        /**
-         * @internal
-         */
-        canManageAnyChannelMembersInCurrentTeam: PropTypes.bool.isRequired,
-
-        /**
-         * The overwritten username that should be shown at the top of the popover
-         */
-        overwriteName: PropTypes.node,
-
-        /**
-         * @internal
-         */
-        actions: PropTypes.shape({
-            getMembershipForEntities: PropTypes.func.isRequired,
-            openDirectChannelToUserId: PropTypes.func.isRequired,
-            openModal: PropTypes.func.isRequired,
-            closeModal: PropTypes.func.isRequired,
-        }).isRequired,
-
-        /**
-         * react-intl helper object
-         */
-        intl: intlShape.isRequired,
-
-        ...Popover.propTypes,
-    }
-
     static defaultProps = {
         isRHS: false,
         hasMention: false,
         status: UserStatuses.OFFLINE,
-        customStatus: {},
-    }
-
-    constructor(props) {
+        customStatus: null,
+    };
+    constructor(props: ProfilePopoverProps) {
         super(props);
         this.state = {
-            loadingDMChannel: -1,
+            loadingDMChannel: undefined,
         };
     }
-
     componentDidMount() {
         const {currentTeamId, userId, channelId} = this.props;
         if (currentTeamId && userId) {
-            this.props.actions.getMembershipForEntities(currentTeamId, userId, channelId);
+            this.props.actions.getMembershipForEntities(
+                currentTeamId,
+                userId,
+                channelId,
+            );
         }
     }
-
-    handleShowDirectChannel = (e) => {
+    handleShowDirectChannel = (e: React.MouseEvent<HTMLAnchorElement>) => {
         const {actions} = this.props;
         e.preventDefault();
-
         if (!this.props.user) {
             return;
         }
-
         const user = this.props.user;
-
-        if (this.state.loadingDMChannel !== -1) {
+        if (this.state.loadingDMChannel !== undefined) {
             return;
         }
-
         this.setState({loadingDMChannel: user.id});
-
-        actions.openDirectChannelToUserId(user.id).then((result) => {
+        actions.openDirectChannelToUserId(user.id).then((result: any) => {
```

Comment:
We can type `result` here with an `error` field

Reply:
result: {error: any}

Reply:
I think `error` is usually of type `ServerError`

Reply:
Done